core/src/main/java/org/bitcoinj/wallet/DefaultCoinSelector.java

```diff
@@ -23,65 +23,50 @@
+    // transaction.
     @Override
     public CoinSelection select(Coin target, List<TransactionOutput> candidates) {
-        ArrayList<TransactionOutput> selected = new ArrayList<>();
-        // Sort the inputs by age*value so we get the highest "coindays" spent.
-        // TODO: Consider changing the wallets internal format to track just outputs and keep them ordered.
-        ArrayList<TransactionOutput> sortedOutputs = new ArrayList<>(candidates);
+        // TODO: Take in network parameters when instantiated, and then test against the current network. Or just have a boolean parameter for "give me everything"
         // When calculating the wallet balance, we may be asked to select all possible coins, if so, avoid sorting
         // them in order to improve performance.
-        // TODO: Take in network parameters when instanatiated, and then test against the current network. Or just have a boolean parameter for "give me everything"
         if (!target.equals(NetworkParameters.MAX_MONEY)) {
-            sortOutputs(sortedOutputs);
-        }
-        // Now iterate over the sorted outputs until we have got as close to the target as possible or a little
-        // bit over (excessive value will be change).
-        long total = 0;
-        for (TransactionOutput output : sortedOutputs) {
-            if (total >= target.value) break;
-            // Only pick chain-included transactions, or transactions that are ours and pending.
-            if (!shouldSelect(output.getParentTransaction())) continue;
-            selected.add(output);
-            total += output.getValue().value;
+            // Sort the inputs by age*value so we get the highest "coindays" spent.
+            // TODO: Consider changing the wallets internal format to track just outputs and keep them ordered.
+            return CoinSelector.sortSelect(target, candidates,  CoinSelector.TXOUT_COMPARATOR, this);
+        } else {
+            return CoinSelector.select(target, candidates, this);
         }
-        // Total may be lower than target here, if the given candidates were insufficient to create to requested
-        // transaction.
-        return new CoinSelection(Coin.valueOf(total), selected);
     }
 
+    @Override
+    public boolean test(TransactionOutput transactionOutput) {
+        // Only pick chain-included transactions, or transactions that are ours and pending.
+        return shouldSelect(transactionOutput.getParentTransaction());
+    }
+
+    /**
+     * Sort a mutable list of {@link TransactionOutput} in place with the {@link CoinSelector#TXOUT_COMPARATOR}.
+     * @param outputs mutable list to be sorted in-place
+     * @deprecated Use functional composition with the static methods in {@link CoinSelector} and/or {@link CoinSelector#TXOUT_COMPARATOR} instead
+     */
+    @Deprecated
     @VisibleForTesting static void sortOutputs(ArrayList<TransactionOutput> outputs) {
```

Comment:
Since this is visible only for testing, we _could_ do away with it entirely. Assuming the `CoinSelector.TXOUT_COMPARATOR` is adequately tested.

Reply:
removed!